util/repeatable_thread.h

```diff
@@ -80,7 +81,17 @@ class RepeatableThread {
       cond_var_.SignalAll();
 #endif
       while (running_) {
+#ifndef NDEBUG
+        if (dynamic_cast<MockTimeEnv*>(env_) != nullptr) {
```

Comment:
I once tried to use `dynamic_cast`, but some builds (don't remember which) failed due to lack of RTTI. Let's see if this can work.